pkg/storage/stores/tsdb/manager.go

```diff
@@ -156,105 +158,123 @@ func (m *tsdbManager) Start() (err error) {
 	return nil
 }
 
-func (m *tsdbManager) BuildFromWALs(t time.Time, ids []WALIdentifier) (err error) {
-	level.Debug(m.log).Log("msg", "building WALs", "n", len(ids), "ts", t)
-	// get relevant wals
-	// iterate them, build tsdb in scratch dir
-	defer func() {
-		m.metrics.tsdbCreationsTotal.Inc()
-		if err != nil {
-			m.metrics.tsdbCreationFailures.Inc()
+func (m *tsdbManager) build(heads *tenantHeads) (err error) {
```

Comment:
Just `build` does not give much context.
```suggestion
func (m *tsdbManager) buildFromHead(heads *tenantHeads) (err error) {
```